### DISCUSSION TITLE
hsr-omniverse simulation setup

### DISCUSSION THREAD

**Discussion by annaliseht** [HumanSupportRobot/QandA#96] [Q&A]
We are attempting to setup the tmc/hsr-omniverse simulator.

We have installed IsaacSim 2022.2.1 on out ubuntu 20.04 machine. We have a nvidia rtx3090 and have the driver version 525.147.05 with CUDA version 12.0 installed. We are getting a large amount of issues that we believe are due to drivers. 

Attached to this question is a file with the output of the command:
`./python.sh hsr-omniverse/sample-ros.py`

Any advice would be appreciated. 
[hsr-omniverse-output.txt](https://github.com/HumanSupportRobot/QandA/files/14519375/hsr-omniverse-output.txt)

**Comment by TomohiroOno33** [comment]
In my experience, it is highly likely that the driver versions other than the recommended ones listed [here](https://docs.omniverse.nvidia.com/isaacsim/latest/installation/requirements.html) will not work.
Could you try to build with Docker?

**Comment by fabiendespinoy** [comment]
First of all, sorry for late reply.

I had quite some issues too. But I confirm that 525 and 545 both works smoothly (on Ubuntu 22.04...).
Here, by looking at your log, I think you may have some unclean install of your NVIDIA drivers.

|---------------------------------------------------------------------------------------------|
| Driver Version: 0             | Graphics API: Vulkan
|=============================================================================================|
| GPU | Name                             | Active | LDA | GPU Memory | Vendor-ID | LUID       |
|     |                                  |        |     |            | Device-ID | UUID       |
|=============================================================================================|
| OS: Linux hulkbuster, Version: 5.15.0-97-generic
| XServer Vendor: The X.Org Foundation, XServer Version: 12013000 (1.20.13.0)
| Processor: Intel(R) Core(TM) i9-7900X CPU @ 3.30GHz | Cores: Unknown | Logical: 20
|---------------------------------------------------------------------------------------------|
| Total Memory (MB): 64172 | Free Memory: 59721
| Total Page/Swap (MB): 2047 | Free Page/Swap: 2047
|---------------------------------------------------------------------------------------------|
2024-03-07 05:54:45 [1,842ms] [Error] [gpu.foundation.plugin] No device could be created. Some known system issues:
- The driver is not installed properly and requires a clean re-install.
- Your GPUs do not support RayTracing: DXR or Vulkan ray_tracing, or hardware is excluded due to performance.
- The driver cannot enumerate any GPU: driver, display, TCC mode or a docker issue. For Vulkan, test it with Vulkaninfo tool from Vulkan SDK, instead of nvidia-smi.
- For Ubuntu, it requires server-xorg-core 1.20.7+ and a display to work without --no-window.
- For Linux dockers, the setup is not complete. Install the latest driver, xServer and NVIDIA container runtime.

I would suggest to try the following instructions:
`sudo apt remove --purge '^nvidia-*.`

Then reboot and clean apt cache, update, and install 525 as suggested by NVIDIA.
Be careful that you may have to reinstall container-toolkit and other NVIDIA tools too.
